proteus/MeshAdaptPUMI/MeshConverter.cpp

```diff
@@ -0,0 +1,412 @@
+#include <algorithm>
+
+#include "MeshAdaptPUMI.h"
+#include "mesh.h"
+#include <apfShape.h>
+
+#include <sstream>
+
+static apf::Numbering* numberOwnedEntitiesFirst(apf::Mesh* m, int dimension)
+{
+  std::stringstream ss;
+  ss << "proteus_number_" << dimension;
+  std::string s = ss.str();
+  apf::FieldShape* shape;
+  if (dimension) /* this switch is just to help rendering */
+    shape = apf::getConstant(dimension);
+  else
+    shape = m->getShape();
+  apf::Numbering* n = createNumbering(m, s.c_str(), shape, 1);
+  apf::MeshEntity* e;
+  apf::MeshIterator* it = m->begin(dimension);
+  int i = 0;
+  while ((e = m->iterate(it)))
+    if (m->isOwned(e))
+      apf::number(n, e, 0, 0, i++);
+  m->end(it);
+  it = m->begin(dimension);
+  while ((e = m->iterate(it)))
+    if (!m->isOwned(e))
+      apf::number(n, e, 0, 0, i++);
+  m->end(it);
+  return n;
+}
+
+//Main API to construct a serial pumi mesh,
+//what we do is contruct the global mesh when working with serial
+//and let Proteus populate the subdomain data structures
+//(though it will be exactly same)
+int MeshAdaptPUMIDrvr::constructFromSerialPUMIMesh(Mesh& mesh)
+{
+  assert(m != 0);
+  std::cout << "Constructing global data structures\n"; 
+
+  int dim = m->getDimension();
+  mesh.nElements_global = m->count(dim);
+
+  mesh.nNodes_global = m->count(0);
+
+  mesh.nElementBoundaries_global = m->count(dim - 1);
+
+  mesh.nEdges_global = m->count(1);
+
+//nNodes_element for now is constant for the entire mesh, Ask proteus about using mixed meshes
+  switch (dim) {
+    case 2:
+      mesh.nNodes_element = 3;
+      mesh.nNodes_elementBoundary = 2;
+      mesh.nElementBoundaries_element = 3;
+      break;
+    case 3:
+      mesh.nNodes_element = 4;
+      mesh.nNodes_elementBoundary = 3;
+      mesh.nElementBoundaries_element = 4;
+      break;
+    default:
+      apf::fail("dimension is not 2 or 3\n");
+      break;
+  }
+
+  std::cerr << "*******Proteus Mesh Stats*********\n";
```

Comment:
It's actually not a duplicate statement for each MPI task. The names of the
variables are misleading. It will print information about the local mesh
part (number of nodes, edges,etc). Either way, I think using a macro will
be a good idea.